packages/graphql/tests/integration/directives/cypher/filtering/cypher-filtering-relationship.int.test.ts

```diff
@@ -0,0 +1,1087 @@
+                    @cypher(
+                        statement: """
+                        MATCH (this)-[:ACTED_IN]->(movie:${Movie})
+                        RETURN movie
+                        """
+                        columnName: "movie"
+                    )
+            }
+        `;
+
+        await testHelper.initNeo4jGraphQL({ typeDefs });
+        await testHelper.executeCypher(
+            `
+            CREATE (m:${Movie} { title: "The Matrix" })
+            CREATE (m2:${Movie} { title: "The Matrix Reloaded" })
+            CREATE (m3:${Movie} { title: "The Matrix Revolutions" })
+            CREATE (a:${Actor} { name: "Keanu Reeves" })
+            CREATE (a)-[:ACTED_IN]->(m)
+            CREATE (a)-[:ACTED_IN]->(m2)
+            CREATE (a)-[:ACTED_IN]->(m3)
+            `,
+            {}
+        );
+
+        const query = /* GraphQL */ `
+            query {
+                ${Movie.plural}(
+                    where: {
+                        actor: {
+                            name: "Keanu Reeves"
+                        } 
+                    }
+                ) {
+                    title
+                    actor {
+                        name
+                    }
+                }
+            }
+        `;
+
+        const gqlResult = await testHelper.executeGraphQL(query);
+
+        expect(gqlResult.errors).toBeFalsy();
+        expect(gqlResult?.data).toEqual({
+            [Movie.plural]: expect.toIncludeSameMembers([
+                {
+                    title: "The Matrix",
+                    actor: {
+                        name: "Keanu Reeves",
+                    },
+                },
+                {
+                    title: "The Matrix Reloaded",
+                    actor: {
+                        name: "Keanu Reeves",
+                    },
+                },
+                {
+                    title: "The Matrix Revolutions",
+                    actor: {
+                        name: "Keanu Reeves",
+                    },
+                },
+            ]),
+        });
+    });
+
+    test("1 to 1 relationship with null filter", async () => {
+        const Movie = testHelper.createUniqueType("Movie");
+        const Actor = testHelper.createUniqueType("Actor");
+
+        const typeDefs = /* GraphQL */ `
+            type ${Movie} @node {
+                title: String
+                released: Int
+                actor: ${Actor}
+                    @cypher(
+                        statement: """
+                        MATCH (this)<-[:ACTED_IN]-(actor:${Actor})
+                        RETURN actor
+                        """
+                        columnName: "actor"
+                    )
+            }
+
+            type ${Actor} @node {
+                name: String
+                movie: ${Movie}
+                    @cypher(
+                        statement: """
+                        MATCH (this)-[:ACTED_IN]->(movie:${Movie})
```

Comment:
Unless I'm misunderstanding, you're pointing out that the test itself is poorly written due to the custom cypher possible returning more than one movie?

I do think there's something to be said about this particular feature being something more for "power users". We can't be sure that the cypher they enter is correct, which is to say the onus is on the user to provide correct data and correct custom cypher.

Please let me know if I've misunderstood your point.

Reply:
Let me rephrase, I believe these tests are wrong or incomplete.
The reason could vary depending on the intention of the feature:
- If we say we cannot handle wrong statements that may return more than a node, then the code should be tested with correct statements (in this case, the statement is wrong as it could return more than a Movie)

- If we say the statement is correct, we should test the behavior when zero or more than a movie is present.

Either way, IMO the cardinality should be handled consistently with the existing behavior, and not depending on whether the target is an Entity or a Scalar or if the field is being used in a filter or a selection